components/formats-gpl/src/loci/formats/in/ZeissLSMReader.java

```diff
@@ -2187,16 +2187,16 @@ private void parseApplicationTags() throws IOException {
           data = in.readString(dataSize);
           break;
         case TYPE_LONG:
-          data = new Integer(in.readInt());
+          data = (Integer) in.readInt();
           break;
         case TYPE_RATIONAL:
-          data = new Double(in.readDouble());
+          data = (Double) in.readDouble();
           break;
         case TYPE_DATE:
-          data = new Long(in.readLong());
+          data = in.readLong();
           break;
         case TYPE_BOOLEAN:
-          data = new Boolean(in.readInt() == 0);
+          data = (Boolean) (in.readInt() == 0);
```

Comment:
Three more unnecessary casts?  Or at least, I don't follow why it's necessary in lines 2190, 2193, 2199, but not in 2196.